### ISSUE TITLE
Roles and permissions

### ISSUE THREAD

**Issue by lcabrini** [AluguntuguiProject/aluguntugui#19]
We need to look into the roles/permissions side of the app, since some functionality should only be limited to certain users. At the moment, the full user list should be limited to a site administrator (this will later have the ability to perform certain operations, such as blocking users).

Let's first look at the categories of users that we need to have:
1. site administrator: a user that can manage other users, languages, etc.
2. user: regular user
3. visitor: not yet registered

I envision that we will later have some sort of "language admin" role as well. That is, a user that is an administrator, but only within one or more languages.

So how do we achieve this? I have come up with the following alternatives.

1. User with id 1 is special. This is similar to how UNIX and Linux work. There is one user with all the powers at their disposal. We could come up with a language admins model later when we need to tackle that problem. The downside here is that site administration is limited to one user only.

2.We create a Role model and use it to create the roles 'site administrator' and 'user'. This would allow for multiple site adminstrators, but still wouldn't effectively solve the problem of language admins, unless you want to create a special role for each language, ie EnglishAdmin, FrenchAdmin, etc. The rights given to each role are more or less hardcoded into the system.

3. A more fine-grained approach using roles and permissions. Here a site admin can create new roles and assign permissions to those roles. This gives a lot of flexibility but at the cost of more complexity.

So what do we do? Option 1 seems more simple, but less flexible. I personally don't like the idea of several physical people sharing the same user account, in the case where you need multiple admins. 

I don't particularly like fine-grained access control. Systems like Drupal just become overly complex because you have to create a role and then click through all the permissions you want to assign to that role. 

In summary, I am mostly in favor of option 2. What do you think?
